### DISCUSSION TITLE
Failing gauge test execution eagerly

### DISCUSSION THREAD

**Discussion by sauravraj007** [getgauge/gauge#1913] [General]
We are running gauge tests on PR builds, we have seen that whenever a scenario/step fails , gauge continues to execute the other scenarios.

Is it possible somehow to fail the gauge test execution eagerly when a scenario fails? I mean I would want to fail the gauge test execution as soon as there is a failure observed in any scenario during its execution , i don't want the rest of the scenarios to get executed.

Is it possible ?

**Comment by sriv** [comment]
No, this isn't possible currently. Gauge is designed for acceptance testing, hence it expects scenarios to be independent flows. Failure in one flow is not assumed to disrupt others. 

Since the cost of running these tests are higher (compared to a unit test), Gauge tries to cover as much as it can in every run.

**Reply by sauravraj007** [reply]
@sriv Thanks for clarification.
